documentation/writing_tests.md

```diff
@@ -0,0 +1,1496 @@
+
+For our Engineering Systems to pick up our tests appropriately, our packages must be configured according to their guidelines. In this section we will go through some of these concepts, and we will provide links that expand them in detail. We will be covering:
+
+- [Engineering goals](#engineering-goals).
+- [CI and nightly test configuration](#ci-and-nightly-test-configuration).
+- [Delivering live tests to our users](#delivering-live-tests-to-our-users).
+
+### Engineering goals
+
+Though the tests must target live resources, we should make sure they only do so when necessary. We must keep in mind the following considerations:
+
+- **Tests should not be flaky.** Tests should pass regardless of who's executing them, when they are running, and how many times they run.
+- **Tests should create the resources they are testing**, either through the code of the tests, or through their configuration files.
+- **The resources created for the tests should be unique.** Running the same test in parallel, multiple times, should not break them.
+- **Avoid calling to timed delays** (like `setTimeout`) to assert that a change happened in the live resources.
+
+Keep in mind that it is valid to have manually created resources during the _development_ of the tests. For example, if you're working with any of the KeyVault clients, you may target a KeyVault created in your personal Azure account. Each one of the individual clients can then assume that a KeyVault provided by environment variables exists, and only focus on creating and managing the resources they're scoped to work with: Keys, Secrets and Certificates. **Ask your team to see if there's a resource already in place for test development.**
+
+Any resource that is not created by the tests must be defined in an [ARM template](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/overview), so that anyone can build a copy of them. This ARM template will be used by the CI pipelines during builds. Under the context of the previous KeyVault example, to automate the creation of the KeyVault that is passed into the tests, it needs to have been defined specifically in the [CI and nightly test configuration](#ci-and-nightly-test-configuration) files. 
+
+Regarding delays and `setTimeout`s, please refer to the [using delays](#using-delays) section.
+
+You'll be able to find a more detailed set of recommendations in the section: [Best Practices for writing tests that target live resources](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/51/Testing-Guidelines) [🔒][TIPS].
+
+### CI and nightly test configuration
+
+To ensure that our tests are executed in the test [Azure DevOps pipelines](https://azure.microsoft.com/en-us/services/devops/pipelines/) that have been previously configured by our team, some configuration files are necessary.
```

Comment:
> previously configured by our team

These pipelines are mostly generated by automation (except for JS live tests pipelines which are manually created because there are good reasons to use per-package pipelines)

Reply:
I'm not sure how to express this without making things more confusing!